### PR TITLE
Update DG

### PR DESCRIPTION
**Pull request by YHWong20**
## Changelog:
* Updated `Acknowledgements` section
* Updated `Setting up, getting started` section
* Updated `Non-functional requirements` section
* Added sequence diagram and description for `Parser` component
* Added sequence diagram and updated description for `Add Item` feature
* Small formatting fixes

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -19,30 +26,70 @@
...
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+[AB3 Developer Guide](https://se-education.org/addressbook-level3/DeveloperGuide.html)
+
+[AB3 GitHub Project Repository](https://github.com/se-edu/addressbook-level3)
 
 ## Setting up, getting started
 
-## Design
+1. Ensure you have Java `11` and above. If you do not have the required version, you can install the Java `11` JDK from this [link](https://www.oracle.com/sg/java/technologies/javase/jdk11-archive-downloads.html).
```
Comment (nur-haziq) [inline docs/DeveloperGuide.md:35]: Thanks for including this portion. Well written! 👍

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -63,37 +110,73 @@ The **Sequence Diagram** below shows how the components interact with each other
...
+
+![ParseSequenceDiagram](images/ParseSequenceDiagram.png)
+
+Here, it is assumed that the user has already provided their input to the `Ui`. 
+Then, `BinBash` retrieves this input and passes it to `Parser`.
+The `Parser` will then process the user input to determine the type of command to be created.
+
+From here, `Parser` will self-call its corresponding `parseXYZCommand()` method.
+Upon calling `parseXYZCommand()`, the `parse()` method of an internal `XYZCommandParser` is invoked, to create the appropriate `Command` (an `XYZCommand` in this case).
+> In some instances, if the command that needs to be created is simple enough (like a `ByeCommand` or `ListCommand`), then `Parser` will directly create the `Command` without the need of an `XYZCommandParser`.
```
Comment (nur-haziq) [inline docs/DeveloperGuide.md:141]: Good clarification.

**Review by nur-haziq — APPROVED** [review]
Sequence diagrams are accurate, and description of the Parser Component is detailed. 😄

File: docs/diagrams/ParseSequenceDiagram.puml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+@startuml
+
+participant ":BinBash"
```
[the remark this answers is not in the document]
Comment (YHWong20) [reply docs/diagrams/ParseSequenceDiagram.puml:3]: Got it, I'll make the changes!

**Pull request merged**